oxen-rust/src/lib/src/core/v_latest/fetch.rs

```diff
@@ -259,13 +303,12 @@ fn collect_missing_entries(
                 "collect_missing_entries for {subtree_paths:?} subtree paths and depth {depth:?}"
             );
             for subtree_path in subtree_paths {
-                let mut unique_hashes = HashSet::new();
                 let Some(tree) = repositories::tree::get_subtree_by_depth_with_unique_children(
                     repo,
                     commit,
                     subtree_path,
-                    Some(&shared_hashes),
-                    Some(&mut unique_hashes),
+                    None,
```

Comment:
This seems wrong to me. I thought the main purpose of this `get_subtree_by_depth_with_unique_children` function was to avoid walking the entire merkle tree for every commit. It seems like we're just disabling this optimization in order to do the aggregation at the top level instead. @rpschoenburg I know you worked on this as well, do you have any input?